packages/core/test/app/utils/number.spec.ts

```diff
@@ -0,0 +1,320 @@
+import '../../setup'
+import { BigNumber, ZERO, ONE } from '../../../src/app/utils'
+import * as assert from 'assert'
+
+describe('BigNumber', () => {
+  describe('constructor', () => {
+    it('should handle number', () => {
+      const num: BigNumber = new BigNumber(10)
+      assert(num.toNumber() === 10, 'Could not construct BigNumber with number')
+    })
+
+    it('should handle base10 string', () => {
+      const num: BigNumber = new BigNumber('10', 10)
+      assert(num.toNumber() === 10, 'Could not construct BigNumber with number')
+    })
+
+    it('should handle hex string', () => {
+      const num: BigNumber = new BigNumber('a', 'hex')
+      assert(num.toNumber() === 10, 'Could not construct BigNumber with number')
+    })
+
+    it('should handle BigNumber', () => {
+      const num: BigNumber = new BigNumber(new BigNumber(10))
+      assert(num.toNumber() === 10, 'Could not construct BigNumber with number')
+    })
+
+    it('should handle Buffer', () => {
+      const num: BigNumber = new BigNumber(new BigNumber(10).toBuffer())
```

Comment:
For completeness we could explicitly test `toBuffer()`

Reply:
Yeah, that may be a good idea. I'm less concerned with what is present in the buffer and more concerned that when we call `.toBuffer()` and then instantiate the result, we get the same value. Might be good to test the different bases, though.